share/poly.go

```diff
@@ -387,6 +387,11 @@ func (p *PubPoly) Equal(q *PubPoly) bool {
 		return false
 	}
 	b := 1
+
+	if len(p.commits) < p.Threshold() || len(q.commits) < p.Threshold() {
```

Comment:
What if `p.Threshold() != q.Threshold()` ?

Reply:
I assumed they just wouldn't be equal, but I guess this might not be the case if `q` has strictly more commits than `p`